### ISSUE TITLE
Hover doesn't always disappear after exit

### ISSUE THREAD

**Issue by aeab13** [JetBrains/compose-jb#1324]
Seems like sometimes the hover exit is not working as expected, as the item is still marked as if it was still being hovered.

Code to reproduce it:

```
fun main() = singleWindowApplication {
    LazyColumn {
        items(count = 10) { i ->
            Text(
                text = "Item number $i",
                modifier = Modifier.clickable {}
            )
        }
    }
}
```
GIF showing the problem where I hover the list's items and sometimes some of them keep being hovered when the mouse is not on top them:

![hover_issue](https://user-images.githubusercontent.com/16917468/139362698-e8e87c46-86e0-4f9c-a11a-4de246f131d4.gif)

OS: Ubuntu 20.04 & Manjaro with KDE
Compose Desktop Version: 1.0.0-beta5 (started happening since the first version that included the hovering)

**Comment by akurasov** [comment]
Managed to reproduce on Mac, but it is really-really difficult

**Comment by aeab13** [comment]
I have noticed that it happens much more in my Manjaro KDE installation than on Ubuntu, not sure why tho. Not sure if it's the desktop environment or there is something different in the build process (like different JDK). In any case, still happens in Ubuntu even if it's less.

**Comment by hrach** [comment]
I have the same problem on Windows. Reproducibility is easy:
```kotlin
@OptIn(ExperimentalAnimationApi::class)
fun main() = singleWindowApplication {
    val interactionSource = MutableInteractionSource()
    val isHovered by interactionSource.collectIsHoveredAsState()
    Box(Modifier.size(100.dp).background(Color.Gray).hoverable(interactionSource)) {
        if (isHovered) {
            Text("Hovered")
        }
    }
}
```

https://user-images.githubusercontent.com/284263/141658149-f00844e3-856b-46c7-b125-5b2b0105f176.mp4

**Comment by igordmn** [comment]
In 1.0.0-rc4 reproduced with scrollbar an heavy lazy list:

https://user-images.githubusercontent.com/5963351/143503791-519ba86f-d053-46c3-b810-d0521f0defa8.mp4

With light lazy list it is hard to reproduce.

**Comment by aeab13** [comment]
With rc5 the issue seems to be solved when simply hovering with the cursor, however, when hovering an item in a lazy list and then scrolling, it still sometimes ends up with a hovered line that shouldn't be.

**Comment by hrach** [comment]
My usecase https://github.com/JetBrains/compose-jb/issues/1324#issuecomment-968131533 behave exactly the same in latest RC5. Windows 11.

**Comment by igordmn** [comment]
> My usecase #1324 (comment) behave exactly the same in latest RC5. Windows 11.

Can you check on 0.0.0-feature-test28112021-dev491? It will be rc6 tomorrow

**Comment by igordmn** [comment]
> My usecase #1324 (comment) behave exactly the same in latest RC5. Windows 11.

Actually, I reproduced it. The issue in your snippet. It should be:
```
val interactionSource = remember { MutableInteractionSource() }
```

**Comment by hrach** [comment]
I really sorry, lame mistake. My non-simiplified usecase works correctly with RC5. THX.

**Comment by aeab13** [comment]
> Can you check on 0.0.0-feature-test28112021-dev491? It will be rc6 tomorrow

Does this version include https://github.com/JetBrains/androidx/pull/135?

**Comment by igordmn** [comment]
> Does this version include JetBrains/androidx#135?

Yes, it includes this fix

**Comment by aeab13** [comment]
The hover issue when scrolling still persists but it's less common.
![list_hover](https://user-images.githubusercontent.com/16917468/143785163-b00ae10e-dbcf-4fd5-83b3-1c92b675cd01.gif)

**Comment by igordmn** [comment]
Can you check if this works in 1.0.0-rc12?

There was one more change regarding this issue.

On my PC it works in all tested cases.

**Comment by aeab13** [comment]
In 1.0.0-rc12 seems to work nicely 🥳 Good job and thanks @igordmn !

**Comment by igordmn** [comment]
It appears, it haven't completely fixed in 1.0.0. The new found cases should be fixed in 1.1.0

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5217">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
